### DISCUSSION TITLE
Issue form preview and validation could be improved

### DISCUSSION THREAD

**Discussion by danielcolinjames** [community/community#4907] [Projects and Issues] [answered]
Love the new issue forms, and the ability to set fields as required is fantastic!

I was helping implement a bug report issue form here https://github.com/helium/explorer/blob/v2/.github/ISSUE_TEMPLATE/bug_report.yml and I couldn't find any easy way to validate the form before it was merged into the main branch of the repo. I didn't even realize there were formatting errors until after it was merged, and I went to /issues/new and it didn't prompt me to choose a template.

So then I went looking for the file in the repo and saw an error message at the top of the file.

Here's the error I saw:
![Screen Shot 2021-08-02 at 11 46 38 AM](https://user-images.githubusercontent.com/10648471/128074002-36a60f8a-5f44-4d40-bc13-1c9de5901535.png)

(Also, that "Learn more" CTA ^ was a dead link and brought me to a 404 page)

After I saw this one, I did a search for "YAML validator" and found this site: http://www.yamllint.com/ which helped me do some trial and error to figure out where the formatting was breaking (I'm new to YAML) instead of doing the trial and error on the repo itself.

Then I still saw this error after merging in a version of the file that the YAML lint website above gave a green light for:
![Screen Shot 2021-08-02 at 11 58 46 AM](https://user-images.githubusercontent.com/10648471/128074326-3a1f32f3-be62-4a62-97bc-aba84f5e8285.png)

And I took a guess and figured that it was talking about a nested field somewhere in the `body` that was set to:
```yaml
value: ""
```
so I deleted that line and merged it, and then everything worked beautifully and the template showed up when creating a new issue.

Maybe I missed it, but some more guidance around how to validate the form before making a pull request would be great, or some sort of "testing" playground sort of like Codepen. All in all I ended up merging 3 different pull requests (which trigger a rebuild of the whole site) to get the formatting right, so it would be great to be able to avoid that. In hindsight, as I'm writing this, I'm realizing I could have just committed changes to the same PR and looked at the file within that PR before merging it to see the errors there, but I still think a validation helper tool of sorts would be super useful.

I'm still a huge fan of this feature though, and I'm sure it's still being worked on, so I hope this is helpful feedback!

**Comment by lukehefson** [comment, marked as answer]
> (Also, that "Learn more" CTA ^ was a dead link and brought me to a 404 page)

Thanks for letting us know @danielcolinjames! 🐛 🔨 

> Maybe I missed it, but some more guidance around how to validate the form before making a pull request would be great, or some sort of "testing" playground sort of like Codepen. All in all I ended up merging 3 different pull requests (which trigger a rebuild of the whole site) to get the formatting right, so it would be great to be able to avoid that.

Your feedback seems similar to https://github.com/github/feedback/discussions/4402 and it’s something we’ve heard from other folk too. Thanks for letting us know your use case it helps us think through the problem and how we might resolve it!
